bitcash/wallet.py

```diff
@@ -714,4 +719,300 @@ def __repr__(self):
+        if self._address is None:
+            self._address = public_key_to_address(self._public_key, version='regtest')
+
+        return self._address
+
+    @property
+    def scriptcode(self):
+        self._scriptcode = (OP_DUP + OP_HASH160 + OP_PUSH_20 +
+                            address_to_public_key_hash(self.address) +
+                            OP_EQUALVERIFY + OP_CHECKSIG)
+        return self._scriptcode
+
+    def to_wif(self):
+        return bytes_to_wif(
+            self._pk.secret,
+            version='regtest',
+            compressed=self.is_compressed()
+        )
+
+    def balance_as(self, currency):
+        """Returns your balance as a formatted string in a particular currency.
+
+        :param currency: One of the :ref:`supported currencies`.
+        :type currency: ``str``
+        :rtype: ``str``
+        """
+        return satoshi_to_currency_cached(self.balance, currency)
+
+    def get_balance(self, currency='satoshi'):
+        """Fetches the current balance by calling
+        :func:`~bitcash.PrivateKeyTestnet.get_unspents`.
+        We do not use `~bitcash.PrivateKeyTestnet.balance_as` as Testnet coins
+        do not have a fiat (e.g. USD) value.
+
+        :param currency: One of the :ref:`supported currencies`.
+        :type currency: ``str``
+        :rtype: ``str``
+        """
+        self.unspents[:] = NetworkAPI.get_unspent_regtest(self.address)
+        self.balance = sum(unspent.amount for unspent in self.unspents)
+        return self.balance
+
+    def get_unspents(self):
+        """Fetches all available unspent transaction outputs.
+
+        :rtype: ``list`` of :class:`~bitcash.network.meta.Unspent`
+        """
+        self.unspents[:] = NetworkAPI.get_unspent_regtest(self.address)
+        self.balance = sum(unspent.amount for unspent in self.unspents)
+        return self.unspents
+
+    def get_transactions(self):
+        """Fetches transaction history.
+
+        :rtype: ``list`` of ``str`` transaction IDs
+        """
+        self.transactions[:] = NetworkAPI.get_transactions_regtest(self.address)
+        return self.transactions
+
+    def create_transaction(self, outputs, fee=None, leftover=None, combine=True,
+                           message=None, unspents=None, custom_pushdata=False):
+        """Creates a signed P2PKH transaction.
+
+        :param outputs: A sequence of outputs you wish to send in the form
+                        ``(destination, amount, currency)``. The amount can
+                        be either an int, float, or string as long as it is
+                        a valid input to ``decimal.Decimal``. The currency
+                        must be :ref:`supported <supported currencies>`.
+        :type outputs: ``list`` of ``tuple``
+        :param fee: The number of satoshi per byte to pay to miners. By default
+                    Bitcash will poll `<https://bitcoincashfees.earn.com>`_ and use a fee
```

Comment:
Should this still be here for regtest?

Reply:
It's not clear what you're referencing. The `create_transaction()` function in the `PrivateKeyRegtest` class (as well as other classes) doesn't have a regtest flag anymore. I just looked at all 31 places the word "regtest" comes up in the wallet.py file though, and all of the current uses of it are necessary (including in the WIF functions).

Reply:
Oh, did you mean the comment messages? To be honest they're just copied from the testnet class and some are probably not related to regtest. I'll take a look through them and adjust accordingly

Reply:
`Bitcash will poll `<https://bitcoincashfees.earn.com>`_ and use a fee` is a vestige from the original codebase and should be removed, I can do that later, don't worry about it @nicolaiskye :)

Reply:
Sounds good!